7.0 causes error when used in project with typescript strict null checks

### Summary

I'm submitting a: <!-- replace '[ ]' with '[x]' where appropriate -->

- [x] bug report
- [ ] feature request
- [ ] question / support request <!-- Are you sure `ng-recaptcha` isthe right place for your request? Consider submitting an issue to this project instead: https://github.com/google/recaptcha -->
- [ ] other

### Description

```sh
> ng build --aot --prod --subresource-integrity

√ Browser application bundle generation complete.

Error: node_modules/ng-recaptcha/recaptcha/recaptcha.component.d.ts:6:59 - error TS2344: Type '(() => void) | undefined' does not satisfy the constraint '(...args: any) => any'.
  Type 'undefined' is not assignable to type '(...args: any) => any'.

6 export declare type RecaptchaErrorParameters = Parameters<ReCaptchaV2.Parameters["error-callback"]>;
                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<!--
Un-comment and fill out the following section if you're submitting a bug (unless the cause is obvious). Try to come up with a reproduction example which demonstrates the bug you're reporting. This will make it easier to reproduce and fix the problem, which makes it more likely to get fixed.
You can use this Stackblitz demo as a starting point: https://stackblitz.com/edit/ng-recaptcha-example
-->

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f17491e0d087ae9ba39cfcc961f1952016b20e75/types/grecaptcha/index.d.ts#L125 defines error callback as `(): void` (this is probably wrong but I am unable to find any documentation to correct it to submit the PR to DT) but the member `"error-callback"` is optional so its type includes `undefined`.

https://github.com/DethAriel/ng-recaptcha/blob/master/src/recaptcha/recaptcha.component.ts#L22-L24 uses this definition.

When the project is included in a project that builds with tsconfig:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "strictNullChecks": true,
```

the above error occurs.

To simplify:
```
// required grecaptcha partial type definition
namespace ReCaptchaV2 {
    export interface Parameters {
        "error-callback"?(): void;
    }
}

// line from recaptcha.component.ts
type RecaptchaErrorParameters = Parameters<ReCaptchaV2.Parameters["error-callback"]>;
```

https://www.typescriptlang.org/play?#code/HYQwtgpgzgDiDGEAEAlCBhEMAu8AWIAagExIDeAUEtUhAB4wD2ATtkgJbDYTMBmCyAAohm4CN2ZRyVGrIBEPZiwC08EABt1AIwQBrOQH4AFAEoAXEgBujdgBMA3DOoBfCq4rYAnjGRo1OfBAAUWYlZmFRSAkpAF4kCLFogB40TACCEgA6BKieKABtBVCVNU0deH0AXQA+eyA

open tsconfig and toggle strictNullChecks to see the error 

Lib versions:

* ng-recaptcha: 7.0.0
* Angular: 11.0.5
* Typescript (`tsc --version`): 4.1.3

--- a/src/recaptcha/recaptcha.component.ts
+++ b/src/recaptcha/recaptcha.component.ts
@@ -19,8 +19,10 @@ import { RECAPTCHA_SETTINGS } from "./tokens";
 
 let nextId = 0;
 
+export type NeverUndefined<T> = T extends undefined ? never : T;
+
 export type RecaptchaErrorParameters = Parameters<
-  ReCaptchaV2.Parameters["error-callback"]
+  NeverUndefined<ReCaptchaV2.Parameters["error-callback"]>
 >;
 
 @Component({

